logviewer/src/main/java/org/togetherjava/logwatcher/oauth/OAuth2LoginConfig.java

```diff
@@ -0,0 +1,93 @@
+import org.springframework.security.oauth2.client.web.AuthenticatedPrincipalOAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+import org.springframework.security.oauth2.core.ClientAuthenticationMethod;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.togetherjava.logwatcher.config.Config;
+
+/**
+ * Configures Spring Security so that we use Discord-OAuth2 as the identity provider
+ */
+@Configuration
+public class OAuth2LoginConfig {
+
+    @Bean
+    public ClientRegistrationRepository clientRegistrationRepository(Config config) {
+        return new InMemoryClientRegistrationRepository(googleClientRegistration(config));
+    }
+
+    @Bean
+    public OAuth2AuthorizedClientService authorizedClientService(
+            ClientRegistrationRepository clientRegistrationRepository) {
+        return new InMemoryOAuth2AuthorizedClientService(clientRegistrationRepository);
+    }
+
+    @Bean
+    public OAuth2AuthorizedClientRepository authorizedClientRepository(
+            OAuth2AuthorizedClientService authorizedClientService) {
+        return new AuthenticatedPrincipalOAuth2AuthorizedClientRepository(authorizedClientService);
+    }
+
+    /**
+     * This is where we actually configure the Security-API to talk to Discord
+     *
+     * @return The ClientRegistration for Discord
+     */
+    private ClientRegistration googleClientRegistration(Config config) {
+        return ClientRegistration.withRegistrationId("Discord")
+            .clientName(config.getClientName())
+            .clientId(config.getClientId())
+            .clientSecret(config.getClientSecret())
+            .clientAuthenticationMethod(ClientAuthenticationMethod.CLIENT_SECRET_POST)
+            .authorizationGrantType(AuthorizationGrantType.AUTHORIZATION_CODE)
+            .scope("identify")
+            .authorizationUri("https://discord.com/api/oauth2/authorize")
+            .userInfoUri("https://discordapp.com/api/users/@me")
+            .userNameAttributeName("username")
+            .tokenUri("https://discordapp.com/api/oauth2/token")
+            .redirectUri(config.getRedirectPath())
+            .build();
+    }
+
+    /**
+     * Configures the Security-API which path's should be protected and generally what to do
+     */
+    @EnableWebSecurity
+    public static class OAuth2LoginSecurityConfig extends VaadinWebSecurityConfigurerAdapter {
+
+        @Override
+        public void configure(WebSecurity web) throws Exception {
+            super.configure(web);
+            web.ignoring().antMatchers("/rest/api/**");
+        }
+
+        @Override
+        protected void configure(HttpSecurity http) throws Exception {
+            //@formatter:off
+            http.authorizeRequests().antMatchers("/rest/api/**").anonymous();
```

Comment:
Why did you turn off the formatter? How does it look with proper formatting?

Reply:
turned it back on, while testing the lines were way longer and the formatter mad it unreadable